pkg/workloads/workloads.go

```diff
@@ -144,16 +144,16 @@ func (tsw *TimeSeriesWrite) NextTokenRange() TokenRange {
 
 func (tsw *TimeSeriesWrite) NextPartitionKey() int64 {
 	tsw.PkPosition += tsw.PkStride
-	if tsw.PkPosition >= tsw.PkCount {
-		tsw.PkPosition = tsw.PkOffset
+	if tsw.PkPosition >= tsw.PkCount - tsw.PkStride  {
```

Comment:
I don't understand this change, how does the thread count related to the partition key ?
how does it stop from creating too many partitions ?

Reply:
the initial value for tsw.PkPosition is pkOffset - pkStride, where pkOffset is threadId pkStride is concurrency. and start value mostly is negative. so when we increase pkPosition on pkstride, we get additional concurency partitions:
example: number of partitions pkCount = 10, concurency = 5. for one of the thread:
1. pkPosistion = -1
2. pkPosssion +=  pkStride(concurency) -> 4
3 if pkPossion >= pkCount is false
4. pkPosision += pkStride(concurency) -> 9
5. if pkPossion >= pkCount is false
so we got additional partition keys.